src/FSharp.MetadataFormat/Main.fs

```diff
@@ -58,21 +143,41 @@ type MemberKind =
 
 type Member =
   { Name : string
+    Attributes : Attribute list
     Category : string
     Kind : MemberKind
     Details : MemberOrValue
     Comment : Comment }
-  static member Create(name, kind, cat, details, comment) =
-    { Member.Name = name; Kind = kind;
+  static member Create(name, attributes, kind, cat, details, comment) =
+    { Member.Name = name; Kind = kind; Attributes = attributes
       Category = cat; Details = details; Comment = comment }
 
-type Type =
+  /// Formats the attributes using the Name. Removes the "Attribute"-suffix by default e.g [<Obsolete>] instead of [<ObsoleteAttribute>]
+  member x.FormatAttributes(?removeAttributeSuffix:bool) =
+    Attribute.FormatAttributes(x.Attributes, ?removeAttributeSuffix = removeAttributeSuffix)
+  /// Formats the attributes using the Name. Removes the "Attribute"-suffix by default e.g [<Obsolete>] instead of [<ObsoleteAttribute>]
+
+  member x.FormatAttributesFullName(?removeAttributeSuffix:bool) =
+    Attribute.FormatAttributesFullName(x.Attributes, ?removeAttributeSuffix = removeAttributeSuffix)
+     
+  /// Gets a value indicating whether this member is obsolete
+  member x.IsObsolete =
+    x.Attributes
+    |> Seq.exists (fun a -> a.IsObsoleteAttribute)
+
+  /// Returns the obsolete message, when this member is obsolete. When its not or no message was specified, an empty string is returned
+  member x.ObsoleteMessage =
+    Attribute.TryGetObsoleteMessage(x.Attributes)
```

Comment:
In-line the required code here.
I see that this is not really "clean" but I have no suggestion on how to improve.

Strictly speaking I'm fine with either way. Note that extensions only work when using regular C# extension methods and might lead to problems with the razor template, so I guess leaving this a member is as good as it gets.